observer/observer.go

```diff
@@ -202,6 +207,34 @@ func (o *Observer) IsAnyJobRunning(repository string) bool {
 	return false
 }
 
+// IsLimitConcurrentJobsReached checks if the limit of concurrent jobs by type (backup, check, etc)
+// has been reached
+func (o *Observer) IsLimitConcurrentJobsReached(jobType cfg.JobType, limit int, repository string) bool {
+	listOfJobs := o.GetJobsByRepository(repository)
```

Comment:
This will only return all the jobs running for this repository. So this function currently checks the limit per type per repository. That's not quite what this feature is about. For example, if we set the backup limit to 10 and for repository a there are 5 running backups and for repository b there are 6 running backups it will still trigger a job for repository c resulting in 12 running backups which would be over the defined limit of 10 backups.

The issue isn't about the amount of jobs that run against one repository but rather how many jobs are running on the cluster and may overload it (IO, memory, network, amount of running pods, etc.)

So the observer needs to be extended so it can return the amount of running jobs by type for all repositories.

Reply:
@Kidswiss I see, I'll change that part then, thanks